### PR TITLE
metrics 

### PR DESCRIPTION
**Pull request by jxmorris12**

### CODE REVIEW

File: textattack/metrics/metrics.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+
+class ModelAccuracy(Metric):
+    key = "Model accuracy"
+
+    @staticmethod
+    def compute(attack_results):
+        total_attacks = len(attack_results)
+        skipped_attacks = TotalSkippedAttacks.compute(attack_results)
+        model_accuracy = (total_attacks - skipped_attacks) * 100.0 / (total_attacks)
```
Comment (uvafan) [inline textattack/metrics/metrics.py:71]: Not introduced by your changes but this actually seems like an unreliable way of computing accuracy, e.g. for targeted classification even if the model is wrong the goal might not be met yet if there are more than 2 classes, so the attack won't be skipped even though it's wrong.

Comment (uvafan) [reply textattack/metrics/metrics.py:71]: Actually this also applied for `AccuracyUnderAttack` to some extent, perhaps we should say that these only make sense for untargeted classification attacks.

File: textattack/metrics/metrics.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                    num_words_changed
+                    * 100.0
+                    / len(result.original_result.attacked_text.words)
+                )
+            else:
+                perturbed_word_percentage = 0
+            perturbed_word_percentages.append(perturbed_word_percentage)
+
+        perturbed_word_percentages = np.array(perturbed_word_percentages)
+        print("perturbed_word_percentages ->", perturbed_word_percentages)
```
Comment (uvafan) [inline textattack/metrics/metrics.py:130]: remove before merging

File: textattack/loggers/attack_log_manager.py [hunk 3]
```diff
@@ -53,112 +85,42 @@ def flush(self):
...
+        #           and
+        #       `log_metrics`
+        # TODO: (maybe) in metrics.py:
+        #   develop a level of abstraction
+        #   that will automatically support percentages/
+        #   averages/etc of different precisions (# decimal
+        #   points) and stuff, and also hopefully reduce
+        #   repeat computations / reuse things better
+        #
+        # TODO: choose smarter default metrics for seq2seq models,
```
Comment (uvafan) [inline textattack/loggers/attack_log_manager.py:109]: How are we going to make the default vary based on model and attack type given this setup?

**Comment by uvafan** [comment]
Another thought: it would be nice if metrics for each sample attacked were in the output csv (when `--enable-csv` toggled of course)

File: textattack/commands/attack/run_attack_single_threaded.py [hunk 4]
```diff
@@ -135,23 +124,21 @@ def run(args, checkpoint=None):
...
-                    args, attack_log_manager, worklist, worklist_tail
+                    args, worklist, worklist_tail, attack_results=attack.attack_results,
                 )
                 new_checkpoint.save()
-                attack_log_manager.flush()
 
         pbar.close()
         print()
         # Enable summary stdout
         if args.disable_stdout:
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply textattack/commands/attack/run_attack_single_threaded.py:137]: no

**Comment by qiyanjun** [comment]
@ @tahmid-kazi Please read through this.

**Comment by qiyanjun** [comment]
@tahmid-kazi can you help summarize?

**Comment by jxmorris12** [comment]
@jinyongyoo Closing this since it's getting close to a year. Please reopen it when you're back to working on it! :)

**Pull request closed without merging**